icu4c/source/tools/toolutil/unewdata.cpp

```diff
@@ -56,15 +56,25 @@ udata_create(const char *dir, const char *type, const char *name,
         *pErrorCode=U_MEMORY_ALLOCATION_ERROR;
         return NULL;
     }
-    
+
+    char dirSepChar = U_FILE_SEP_CHAR;
+#if (U_FILE_SEP_CHAR != U_FILE_ALT_SEP_CHAR)
+    // We may need to append a different directory separator when building for Cygwin or MSYS2.
+    if(dir && *dir) {
+      if(!uprv_strchr(dir, U_FILE_SEP_CHAR) && uprv_strchr(dir, U_FILE_ALT_SEP_CHAR)) {
```

Comment:
Note: There might be a preference to be explicit and check for `!= nullptr` and `== nullptr` here.

However, the ICU code is somewhat [inconsistent](https://github.com/unicode-org/icu/blob/master/icu4c/source/common/locid.cpp#L402) about this, so I don't have strong opinion on it